### PR TITLE
Set --cpus-per-task=$SLURM_CPUS_ON_NODE for srun starting orted.

### PR DESCRIPTION
**Pull request by cinek810**

### CODE REVIEW

**Review by rhc54 — APPROVED** [review]
Makes sense - thx!

File: src/mca/plm/slurm/plm_slurm_module.c [hunk 1]
```diff
@@ -270,6 +271,13 @@ static void launch_daemons(int fd, short args, void *cbdata)
     /* start one orted on each node */
     prte_argv_append(&argc, &argv, "--ntasks-per-node=1");
 
+    /* add all CPUs to this task */
+    cpus_on_node = getenv("SLURM_CPUS_ON_NODE");
+    if(cpus_on_node) {
+	    asprintf(&tmp, "--cpus-per-task=%s", cpus_on_node);
+	    opal_argv_append(&argc, &argv, tmp);
+    }
```
Comment (rhc54) [inline src/mca/plm/slurm/plm_slurm_module.c:279]: Missed one thing - need to `free(tmp);` after the append as the `opal_argv_append` function copies the string.

Comment (cinek810) [reply src/mca/plm/slurm/plm_slurm_module.c:279]: Added.

Comment (alansill) [reply src/mca/plm/slurm/plm_slurm_module.c:280]: TravisCI says:
"plm_slurm_module.c:278:6: error: implicit declaration of function
6466      'opal_argv_append' is invalid in C99
6467      [-Werror,-Wimplicit-function-declaration]
6468            opal_argv_append(&argc, &argv, tmp);
6469            ^
6470plm_slurm_module.c:278:6: note: did you mean 'prte_argv_append'?"

Comment (rhc54) [reply src/mca/plm/slurm/plm_slurm_module.c:280]: Yeah, I missed that one - shifting back/forth between code bases too much. It should definitely be "prte_argv_append"

Comment (rhc54) [reply src/mca/plm/slurm/plm_slurm_module.c:280]: Thanks @cinek810

**Pull request merged**